### PR TITLE
TF-2387 Starting page for mobile

### PR DESCRIPTION
**Pull request by dab246**
## Issue 

#2387 

## Demo

https://github.com/linagora/tmail-flutter/assets/80730648/7d3acf88-61d7-4d32-81c0-41da7a5658bc

### CODE REVIEW

**Comment by hoangdat** [comment]
LGTM, but building failed

**Review by nqhhdev — APPROVED** [review]

File: .github/workflows/image.yaml
Comment (tk-nguyen) [inline .github/workflows/image.yaml:1]: Also need to add the ssh config to the `Build and push image` step
```diff
diff --git a/.github/workflows/image.yaml b/.github/workflows/image.yaml
index 542efc27b..b5a5149b2 100644
--- a/.github/workflows/image.yaml
+++ b/.github/workflows/image.yaml
@@ -56,6 +56,8 @@ jobs:
             type=gha
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+          ssh: |
+            default
 
   build-release-image:
     name: Build release image
@@ -106,3 +108,5 @@ jobs:
             type=gha
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+          ssh: |
+            default

```

File: Dockerfile [hunk 1]
```diff
@@ -8,8 +8,14 @@ WORKDIR /app
 
 COPY . .
 
+RUN apt update && \
+    apt install openssh-client -y && \
+    rm -rf assets/js/* && \
```
Comment (tk-nguyen) [inline Dockerfile:13]:
```suggestion
```
This line can be deleted, we are not building `libolm`.

File: Dockerfile [hunk 2]
```diff
@@ -8,8 +8,14 @@ WORKDIR /app
...
+    mkdir ~/.ssh && \
+    ssh-keyscan github.com >> ~/.ssh/known_hosts
+COPY --from=olm-builder /result/javascript assets/js/package
```
Comment (tk-nguyen) [inline Dockerfile:16]:
```suggestion
```
Same as above.

File: .github/workflows/image.yaml
Comment (dab246) [reply .github/workflows/image.yaml:1]: done

File: Dockerfile (hunk 1 above)
Comment (dab246) [reply Dockerfile]: done

File: Dockerfile (hunk 2 above)
Comment (dab246) [reply Dockerfile]: done

File: Dockerfile [hunk 3]
```diff
@@ -8,8 +8,12 @@ WORKDIR /app
 
 COPY . .
 
+RUN apt update && \
+    apt install openssh-client -y && \
+    mkdir ~/.ssh && \
+    ssh-keyscan github.com >> ~/.ssh/known_hosts
 # Precompile tmail flutter
-RUN bash prebuild.sh
+RUN --mount=type=ssh,required=true ./prebuild.sh
```
Comment (tk-nguyen) [inline Dockerfile:16]:
```suggestion
RUN --mount=type=ssh,required=true bash prebuild.sh
```
It should be prefixed with bash. Else you can add execute permission to the script and commit the change:
```
chmod +x prebuild.sh
```

**Pull request merged**